src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -94,6 +96,8 @@ private object DeserializeObject(int typeIndex)
             "Custom readers as well as custom objects on the resources file are not observable by the trimmer and so required assemblies, types and members may be removed.")]
         private bool InitializeBinaryFormatter()
         {
+            Debug.Assert(Monitor.IsEntered(this));
```

Comment:
Same here. That way in the future people that look at this code will know why is the protection expected.

Reply:
Removed this one too. Not needed.